source/js/components/Dashboard/ResultView/ResultList.jsx

```diff
@@ -144,7 +143,8 @@ class ResultList extends Component {
             { sortedItems.map((item) => (
               <ResultListItem
                 item={ item }
-                categoryColor={ getCategoryColor(item) }
+                categoryColor={ getCategory(item).get('color') }
+                editLinkText={ getCategory(item).get('editLinkText') }
```

Comment:
Since the number of categories low, the performance penalty shouldn't be very high. Nevertheless, I adapted the code and rebased the commit.